components/builder-api/src/depot/server.rs

```diff
@@ -786,6 +786,15 @@ fn upload_package(req: &mut Request) -> IronResult<Response> {
         checksum_from_param, ident
     );
 
+    // This stinks but is necessary since memcached Client returns mutable reference
+    // and iron chain.link returns everything wrapped in Arc
+    // The tradeoff here is that we have to initialize a new client on every request until
+    // we can switch to a framework that's not Iron
+    let memcache_host_strings: Vec<String> =
+        depot.memcached.iter().map(|h| format!("{}", h)).collect();
+    let memcache_hosts: Vec<&str> = memcache_host_strings.iter().map(AsRef::as_ref).collect();
+    let mut client = memcache::Client::new(memcache_hosts).unwrap();
```

Comment:
Ugh... yeah we can live with it, we're close to landing the new framework..

Reply:
![](https://media2.giphy.com/media/3ohs7Ys9J8XyFVheg0/200w.gif)